src/nvim/ex_docmd.c

```diff
@@ -9952,3 +9972,100 @@ bool cmd_can_preview(char_u *cmd)
 
   return false;
 }
+
+/// Return the list of the commands in the specified buffer,
+/// if no buffer is specified, then return list of global(non-buffer) the
+/// commands
+///
+/// @param[in] buf the specified buffer, if NULL then global commands required
+///
+/// @return The array of the commands and the information of each command
+ArrayOf(Dictionary) commands_array(buf_T *buf)
+{
+  Array rv = ARRAY_DICT_INIT;
+  int i = 0, j = 0;
+  char Arg[2] = "";
+  garray_T *gap;
+  if (buf == NULL) {
+    gap = &ucmds;
+  } else {
+    gap = &buf->b_ucmds;
+  }
+  for (i = 0; i < gap->ga_len; i++) {
+    Dictionary dic = ARRAY_DICT_INIT;
+    bool ck = false;
+    char Range[10] = "";
+    ucmd_T *cmd = USER_CMD_GA(gap, i);
+
+    // Name
+    PUT(dic, "name", STRING_OBJ(cstr_to_string((const char *)cmd->uc_name)));
+
+    // Argument
+    switch (cmd->uc_argt & (EXTRA|NOSPC|NEEDARG)) {
+    case 0:                    Arg[0] = '0'; break;
+    case(EXTRA):               Arg[0] = '*'; break;
+    case(EXTRA|NOSPC):         Arg[0] = '?'; break;
+    case(EXTRA|NEEDARG):       Arg[0] = '+'; break;
+    case(EXTRA|NOSPC|NEEDARG): Arg[0] = '1'; break;
+    }
+    PUT(dic, "argument", STRING_OBJ(cstr_to_string((const char *)Arg)));
+
+    // Definition
+    PUT(dic, "definition",
+        STRING_OBJ(cstr_to_string((const char *)cmd->uc_rep)));
+
+    // Complete
+    char *cmd_compl = get_command_complete(cmd->uc_compl);
+    if (cmd_compl == NULL) {
+      PUT(dic, "complete",
```

Comment:
typically in situations like this we just omit the key.

Reply:
OK, sure, done